armi/nuclearDataIO/cccc/tests/test_dif3d.py

```diff
@@ -0,0 +1,146 @@
+
+import os
+import unittest
+
+from armi.nuclearDataIO.cccc import dif3d
+from armi.utils.directoryChangers import TemporaryDirectoryChanger
+
+THIS_DIR = os.path.dirname(__file__)
+
+SIMPLE_HEXZ_INP = os.path.join(THIS_DIR, "../../tests", "simple_hexz.inp")
+SIMPLE_HEXZ_DIF3D = os.path.join(THIS_DIR, "fixtures", "simple_hexz.dif3d")
+
+
+class TestDif3d(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        """Load DIF3D data from binary file."""
+        cls.df = dif3d.Dif3dStream.readBinary(SIMPLE_HEXZ_DIF3D)
+
+    def test__rwFileID(self):
+        """Verify the file identification info."""
+        self.assertEqual(self.df.metadata["HNAME"], "DIF3D")
+        self.assertEqual(self.df.metadata["HUSE1"], "")
+        self.assertEqual(self.df.metadata["HUSE2"], "")
+        self.assertEqual(self.df.metadata["VERSION"], 1)
+
+    def test__rwFile1DRecord(self):
+        """Verify the rest of the metadata"""
+        title = "3D Hex-Z to generate NHFLUX file"
+
+        EXPECTED_TITLE = ["3D Hex", "-Z to", "genera", "te NHF", "LUX fi", "le"] + [
+            "" for i in range(5)
+        ]
+        for i in range(10):
+            self.assertEqual(self.df.metadata[f"TITLE{i}"], EXPECTED_TITLE[i])
+        self.assertEqual(self.df.metadata["MAXSIZ"], 10000)
+        self.assertEqual(self.df.metadata["MAXBLK"], 1800000)
+        self.assertEqual(self.df.metadata["IPRINT"], 0)
+
+    def test__rw2DRecord(self):
+        """Verify the Control Parameters"""
+        EXPECTED_2D = [
+            0,
+            0,
+            0,
+            10000,
+            30,
+            0,
+            1000000000,
+            5,
+            0,
+            0,
+            50,
+            0,
+            1,
+            1,
+            0,
+            0,
+            0,
+            110,
+            10,
+            100,
+            1,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            10,
+            40,
+            32,
+            0,
+            0,
+            2,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+            0,
+        ]
+        for i, param in enumerate(dif3d.FILE_SPEC_2D_PARAMS):
+            self.assertEqual(self.df.twoD[param], EXPECTED_2D[i])
+
+    def test__rw3DRecord(self):
+        """Verify the Convergence Criteria"""
+        EXPECTED_3D = [
+            1e-7,
+            1e-5,
+            1e-5,
+            3.823807613470224e-01,
+            1e-3,
+            4e-2,
+            1e0,
+            0e0,
+            0e0,
+            9.999999747378752e-05,
+        ] + [0.0 for i in range(1, 21)]
+        for i, param in enumerate(dif3d.FILE_SPEC_3D_PARAMS):
+            self.assertEqual(self.df.threeD[param], EXPECTED_3D[i])
+
+    def test__rw4DRecord(self):
```

Comment:
Same issue here as I state below for 5D. We could find a way to better test this, but I don't think it's critical because I don't think this feature is active used.

Reply:
attempting to test now

Reply:
added